nav2-usability/scenario-code/deeplink-pathparam/router.dart

```diff
@@ -18,6 +15,38 @@ class Book {
   Book(this.title, this.author);
 }
 
+class AppState extends ChangeNotifier {
+  final List<Book> _books = [
+    Book('Stranger in a Strange Land', 'Robert A. Heinlein'),
+    Book('Foundation', 'Isaac Asimov'),
+    Book('Fahrenheit 451', 'Ray Bradbury'),
+  ];
+
+  int? _selectedBookId;
+
+  int? get selectedBookId => _selectedBookId;
```

Comment:
I know we are not necessary follow the same style in flutter/flutter, but we usually use this order
```
int? get selectedBookId => _selectedBookId;
int? _selectedBookId;
set selectedBookId(int? id) {
    _selectedBookId = id;
    notifyListeners();
  }
```
I will let you decide whether we want to follow it here